homeassistant/components/comelit/alarm_control_panel.py

```diff
@@ -0,0 +1,148 @@
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Comelit VEDO system alarm control panel devices."""
+
+    coordinator: ComelitVedoSystem = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ComelitAlarmEntity(coordinator, device, config_entry.entry_id)
+        for device in coordinator.data[ALARM_AREAS].values()
+    )
+
+
+class ComelitAlarmEntity(CoordinatorEntity[ComelitVedoSystem], AlarmControlPanelEntity):
+    """Representation of a Ness alarm panel."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_code_format = CodeFormat.NUMBER
+    _attr_code_arm_required = False
+    _attr_supported_features = (
+        AlarmControlPanelEntityFeature.ARM_AWAY
+        | AlarmControlPanelEntityFeature.ARM_HOME
+    )
+
+    def __init__(
+        self,
+        coordinator: ComelitVedoSystem,
+        area: ComelitVedoAreaObject,
+        config_entry_entry_id: str,
+    ) -> None:
+        """Initialize the alarm panel."""
+        self._api = coordinator.api
+        self._area_index = area.index
+        super().__init__(coordinator)
+        # Use config_entry.entry_id as base for unique_id
+        # because no serial number or mac is available
+        self._attr_unique_id = f"{config_entry_entry_id}-{area.index}"
+        self._attr_device_info = coordinator.platform_device_info(area)
+        if area.p2:
+            self._attr_supported_features |= AlarmControlPanelEntityFeature.ARM_NIGHT
+
+    @property
+    def _area(self) -> ComelitVedoAreaObject:
+        """Return area object."""
+        return self.coordinator.data[ALARM_AREAS][self._area_index]
+
+    @property
+    def available(self) -> bool:
+        """Return True if alarm is available."""
+        if self._area.human_status in [AlarmAreaState.ANOMALY, AlarmAreaState.UNKNOWN]:
+            return False
+        return super().available
+
+    @property
+    def state(self) -> StateType:
+        """Return the state of the alarm."""
+
+        _LOGGER.debug(
+            "Area %s status is: %s. Armed is %s",
+            self._area.name,
+            self._area.human_status,
+            self._area.armed,
+        )
+        if self._area.human_status == AlarmAreaState.ARMED:
+            if self._area.armed == ALARM_AREA_ARMED_STATUS["away"]:
+                return STATE_ALARM_ARMED_AWAY
+            if self._area.armed == ALARM_AREA_ARMED_STATUS["night"]:
+                return STATE_ALARM_ARMED_NIGHT
+            return STATE_ALARM_ARMED_HOME
+
+        if self._area.human_status == AlarmAreaState.DISARMED:
+            return STATE_ALARM_DISARMED
+        if self._area.human_status == AlarmAreaState.ENTRY_DELAY:
+            return STATE_ALARM_DISARMING
+        if self._area.human_status == AlarmAreaState.EXIT_DELAY:
+            return STATE_ALARM_ARMING
+        if self._area.human_status == AlarmAreaState.TRIGGERED:
+            return STATE_ALARM_TRIGGERED
```

Comment:
Can we maybe use a reverse map for this?

so it would be like

`ALARM_AREA_STATES[self._area.human_status]`

Reply:
Dont' know how to handle `AlarmAreaState.ARMED` as there are 3 sub-types.

Reply:
You can at least change these 4 to using the dict